examples/filter_function_items_before_line_10.rs

```diff
@@ -0,0 +1,6 @@
+use tree_sitter::Node;
+
+#[no_mangle]
+pub extern "C" fn filterer(node: &Node) -> bool {
+    node.kind() == "function_item" && node.start_position().row < 10
```

Comment:
Performance-wise I didn't try anything that spit out an insane # of matches but for something like this that produces a reasonable # of matches I am seeing it run almost as fast vs the equivalent query + filter combo*

Which seems insane, but classic incorrect guess about optimization/performance I guess

[*] if you set the `--language` flag that is - there seems to be some kind of performance issue with the combination of no-query + auto-language. Which also seems to go against my mental model of "the only known file-types it should be walking over is just Rust files which is what happens with passing `--language rust` so should be basically exactly the same"?

Reply:
Hmm, yeah I don't really understand that.